qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,380 @@
+        in the following code block.
+
+    .. code-block:: python
+
+        from qiskit.algorithms.optimizers import OptimizerResult
+
+        def my_minimizer(fun, x0, jac=None, bounds=None) -> OptimizerResult:
+            # Note that the callable *must* have these argument names!
+            # Args:
+            #     fun (callable): the function to minimize
+            #     x0 (np.ndarray): the initial point for the optimization
+            #     jac (callable, optional): the gradient of the objective function
+            #     bounds (list, optional): a list of tuples specifying the parameter bounds
+
+            result = OptimizerResult()
+            result.x = # optimal parameters
+            result.fun = # optimal function value
+            return result
+
+    The above signature also allows one to directly pass any SciPy minimizer, for instance as
+
+    .. code-block:: python
+
+        from functools import partial
+        from scipy.optimize import minimize
+
+        optimizer = partial(minimize, method="L-BFGS-B")
+
+    Attributes:
+        estimator: The estimator primitive to compute the expectation value of the circuits.
+        ansatz: The parameterized circuit used as an ansatz for the wave function.
+        optimizer: A classical optimizer to find the minimum energy. This can either be a
+            Qiskit :class:`.Optimizer` or a callable implementing the :class:`.Minimizer` protocol.
+        gradient: An optional gradient function or operator for the optimizer.
+        initial_point: An optional initial point (i.e. initial parameter values) for the optimizer.
+            If not provided, a random initial point with values in the interval :math:`[0, 2\pi]`
+            is used.
+
+    References:
+        [1] Peruzzo et al, "A variational eigenvalue solver on a quantum processor"
+            `arXiv:1304.3061 https://arxiv.org/abs/1304.3061>`_
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        ansatz: QuantumCircuit | None = None,
+        optimizer: Optimizer | Minimizer | None = None,
+        gradient: BaseEstimatorGradient | None = None,
+        initial_point: Sequence[float] | None = None,
+        callback: Callable[[int, np.ndarray, float, float], None] | None = None,
+    ) -> None:
+        """
+        Args:
+            estimator: The estimator primitive to compute the expectation value of the circuits.
+            ansatz: A parameterized circuit, preparing the ansatz for the wave function. If not
+                provided, this defaults to a :class:`.RealAmplitudes` circuit.
+            optimizer: A classical optimizer to find the minimum energy. This can either be a
+                Qiskit :class:`.Optimizer` or a callable implementing the :class:`.Minimizer`
+                protocol. Defaults to :class:`.SLSQP`.
+            gradient: An optional gradient function or operator for the optimizer. initial_point: An
```

Comment:
The initial_point Arg seems to have wrapped on the end.

gradient here is BaseEstimatorGradient. In current VQE it was an opflow gradient or a callable. Whether we want to allow callable - it has to match what the optimizer native consumes - its easy enough to allow a path like was done before. The text for gradient needs improvement - maybe optional gradient function related to the callable - operator seems odd, the gradients then were part of opflow...